src/DebugChannel.js

```diff
@@ -42,14 +44,36 @@ export class DebugChannel {
   /* Request object based on the protocol
   */
   readIn(): string {
-    let contents = "";
-    while (contents.length === 0) {
-      contents = this.fs.readFileSync(this.inFilePath, "utf8");
+    let message = "";
+    while (true) {
+      let contents = this.fs.readFileSync(this.inFilePath, "utf8");
+
+      if (contents.length === 0) {
+        continue;
+      }
+      // format: <length>--<contents>
+      let separatorIndex = contents.indexOf(TWO_DASH);
+      if (separatorIndex === -1) {
+        continue;
+      }
+
+      let messageLength = parseInt(contents.slice(0, separatorIndex), 10);
+      if (isNaN(messageLength)) {
+        continue;
+      }
+
+      let startIndex = separatorIndex + TWO_DASH.length;
+      let endIndex = separatorIndex + TWO_DASH.length + messageLength;
+      message = contents.slice(startIndex, endIndex);
+      if (message.length < messageLength) {
+        continue;
+      }
+      break;
     }
     //clear the file
     this.fs.writeFileSync(this.inFilePath, "");
     this.requestReceived = true;
-    return contents;
+    return message;
```

Comment:
I remember, per original design, we need to send an acknowledge response to UI once successfully read a request so that UI know it is safe to send another request. Is this in future PR or we changed this design?

Reply:
The response/acknowledgement will be sent next time Prepack is stopped, which may be immediately or after running for a while, so we may not know the know the response at the time of reading in. The caller (or something else later) is responsible for sending a response. I will add a comment here.